subprojects/base-services/src/main/java/org/gradle/internal/work/WorkerLeaseService.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.work;
+
+import org.gradle.internal.concurrent.Stoppable;
+import org.gradle.internal.resources.ProjectLeaseRegistry;
+import org.gradle.internal.resources.ResourceLock;
+
+import java.util.concurrent.Executor;
+
+public interface WorkerLeaseService extends WorkerLeaseRegistry, ProjectLeaseRegistry, Stoppable {
```

Comment:
I wonder if there really is value in having the subinterfaces like WorkerLeaseRegistry or ProjectLeaseRegistry anymore? Since we want to make the interface more generic anyway, I feel like having a single `LeaseService` is probably the simplest to understand.